model/minsky.cc

```diff
@@ -713,6 +725,103 @@ namespace minsky
        });
     canvas.requestRedraw();
   }
+  
+  /*
+  For using GSL non-linear least square fitting routines
+  patterned on first Large Nonlinear Least Squares Example.
+  See online documentation at https://www.gnu.org/software/gsl/doc/html/nls.html
+  */
+      
+namespace
+{	
+
+   int residual_func(const gsl_vector * x, void *params,
+          gsl_vector * f)
+   {
+	if (params==NULL) return GSL_EBADFUNC;
+	else { 	  
+		
+	  struct OPTpars *par = ((struct OPTpars *)params); 
```

Comment:
The modern way is to write this as 

`auto par=reinterpret_cast<OPTPars*>(params);`

Reply:
I have used par = static_cast<OptParams*>(params), if that is okay?